src/components/input-panel/spec-editor/renderer.tsx

```diff
@@ -102,6 +102,10 @@ class Editor extends React.Component<Props, {}> {
 
     if (this.props.history.location.pathname.indexOf('/edited') === -1) {
       this.props.history.push('/edited');
+      window.onbeforeunload =e => {
```

Comment:
Did you try this? I think this always catches, not just when we have unsaved changes. Please test carefully.

Reply:
@domoritz  Sir , you are right ! . It will alert the user whether the user have unsaved changes or not . I thought that we have alerted the user that  " Any unsaved changes will be lost? " and user will handle this query manually as he will also get assured also once if he/she have saved the changes . If you want dialog box only to appear when the changes are unsaved , I will try to FIX it asap . But Please don't mind i was thinking we should ask atleast once before leaving the editor as every editor site does .

Reply:
I see. I think it's better if we only show a warning if the user actually loses information if they navigate away. Thanks for looking into this!

Reply:
@domoritz ok i figured out that how we are storing last state and if we close the editor , on again opening the editor it renders last state in input panel . But it has led us to a new bug which is with the "Browser Back Button" . Let's say a user edited twice and than he jumps back to his previous edit using back button , there is no way he can get his edited code back by using forward button . As you know we are only storing the last state and it gets updated on every edit , we are not preserving history of multiple edits . i hope you are getting what i am trying to convey . Thanks for consideration !

Reply:
Yes, there are numerous usability issues with back and forward buttons but fixing them requires some major changes that we have not done yet. You're more than welcome to tackle this but it's not an easy task.

Reply:
Yes , its seems to be a tricky task , i will keep trying if i figured out any solution , i will try to keep you updated on this bug !